forc-plugins/forc-crypto/src/keys/vanity.rs

```diff
@@ -0,0 +1,551 @@
+use fuel_crypto::{fuel_types::Address, PublicKey, SecretKey};
+use fuels_accounts::wallet::{generate_mnemonic_phrase, DEFAULT_DERIVATION_PATH_PREFIX};
+use fuels_core::types::bech32::{Bech32Address, FUEL_BECH32_HRP};
+use rayon::iter::{self, Either, ParallelIterator};
+use regex::Regex;
+use serde_json::json;
+use std::{
+    error::Error,
+    fmt, fs,
+    path::PathBuf,
+    time::{Duration, Instant},
+};
+use tokio::runtime::Runtime;
+
+forc_util::cli_examples! {
+    crate::Command {
+        [ Generate a checksummed vanity address with a given prefix => "forc crypto vanity --starts-with aaa" ]
+    }
+}
+
+#[derive(Debug, clap::Parser)]
+#[clap(
+    version,
+    about = "Generate a vanity address",
+    after_help = "Generate vanity addresses for the Fuel blockchain"
+)]
+pub struct Arg {
+    /// Prefix regex pattern or hex string.
+    #[arg(long, value_name = "PATTERN", required_unless_present = "ends_with")]
+    pub starts_with: Option<String>,
```

Comment:
Related to my other comment, since we aren't really supporting regex I think this should be:

```suggestion
    /// Desired hex sting prefix.
    #[arg(long, value_name = "HEX_STRING", required_unless_present = "ends_with")]
    pub starts_with: Option<String>,
```

And the same for ends_with.

Reply:
addressed: https://github.com/FuelLabs/sway/pull/6661/commits/6746af46c48b9873e9a649fee075e7147e6069fd